src/main/java/cuchaz/enigma/gui/DecompiledClassSource.java

```diff
@@ -73,18 +72,13 @@ private String remapToken(Deobfuscator deobfuscator, Token token, Token movedTok
 		return null;
 	}
 
-	@Nullable
-	private String proposeName(Deobfuscator deobfuscator, Entry<?> entry) {
-		if (entry instanceof FieldEntry) {
-			for (EnigmaPlugin plugin : deobfuscator.getPlugins()) {
-				String owner = entry.getContainingClass().getFullName();
-				String proposal = plugin.proposeFieldName(owner, entry.getName(), ((FieldEntry) entry).getDesc().toString());
-				if (proposal != null) {
-					return proposal;
-				}
-			}
-		}
-		return null;
+	private Optional<String> proposeName(Deobfuscator deobfuscator, Entry<?> entry) {
+		Stream<String> proposals = deobfuscator.getPlugins()
+				.map(plugin -> plugin.proposeName(entry, deobfuscator.getMapper()))
+				.filter(Optional::isPresent)
+				.map(Optional::get);
```

Comment:
If we use nullable strings, you just need
```java
.filter(Objects::nonNull)
```
instead of 
```java
.filter(Optional::isPresent)
.map(Optional::get)
```

Reply:
If Enigma is using Java 9, this can be replaced with
```java
.flatMap(Optional::stream)
```

Reply:
we are still on java 8, unfortunately.

Reply:
Many users will also be. We're not migrating any of these projects to Java 9+, and if we do, it will be to Java 11 or higher directly.

Reply:
Well then it can be replaced with
```java
.flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
```

Reply:
This is too heavyweight compared to a simple ispresent and get pipelines

Reply:
It is less heavyweight than two stream calls, I think? Use of `Optional#get` is strongly discouraged.